flink-core/src/main/java/org/apache/flink/api/common/BatchShuffleMode.java

```diff
@@ -41,7 +41,7 @@
  * to run upstream and downstream tasks simultaneously.
  */
 @PublicEvolving
-public enum ShuffleMode implements DescribedEnum {
+public enum BatchShuffleMode implements DescribedEnum {
```

Comment:
I thought it might be helpful information for users to properly understand the difference between streaming mode and batch mode. But I can remove it if you think it is rather confusing.

Reply:
I agree the information can be useful for understanding. 
I was a bit confused because `streaming`/`batch` modes stands for `streaming`/`batch` jobs. But this enum only works for batch jobs.
So maybe explain it against `PIPELINED`/`BLOCKING` exchanges instead of `streaming`/`batch` modes?

Reply:
I removed the mentioning of streaming/batch mode in the enum values. But I would leave the more extensive class JavaDoc to understand the meaning of `BatchShuffleMode` in general and why it doesn't apply to streaming mode.

Reply:
> The shuffle mode depends on the configured {@link ExecutionOptions#RUNTIME_MODE} 

This seems to be outdated.

> is only relevant for batch executions on bounded streams.

I do not quite understand what it. Could you explain it a bit more?

> I would leave the more extensive class JavaDoc to understand the meaning of BatchShuffleMode in general and why it doesn't apply to streaming mode.

I agree the more extensive class JavaDoc is good. But currently it says that streaming mode uses pipelined and batch mode uses blocking,  which is actually against the goal that we want shuffle mode of batch mode to be configurable.
For example, maybe we can change "In streaming mode" to "With pipelined exchanges", and later say that streaming job always uses this exchange and batch job can also use it to achieve low latency or higher throughput.

Reply:
I updated the PR. I hope I could address your concerns.

Reply:
Thanks for the update. It now looks generally good to me, except for the part for "low latency" and "higher throughput".
When saying "low latency" and "higher throughput" in my last comment, I actually means the benefit of pipelined exchanges. Latency does not make much sense for batch jobs so I used "higher throughput" instead. 
I think we can simply remove "Batch jobs can be configured for lower latency or higher throughput" and "In order to achieve higher throughput".
Sorry for the confusion caused by my words.